packages/shared/src/hooks/useConditionalFeature.ts

```diff
@@ -0,0 +1,44 @@
+import { useQuery } from '@tanstack/react-query';
+import { JSONValue, WidenPrimitives } from '@growthbook/growthbook';
+import { useContext } from 'react';
+import { generateQueryKey, RequestKey } from '../lib/query';
+import { Feature } from '../lib/featureManagement';
+import { disabledRefetch } from '../lib/func';
+import AuthContext from '../contexts/AuthContext';
+import { useGrowthBookContext } from '../components/GrowthBookProvider';
+
+interface UseConditionalFeature<T> {
+  value: WidenPrimitives<T>;
+  isLoading: boolean;
+}
+export const useConditionalFeature = <T extends JSONValue>({
+  feature,
+  shouldEvaluate,
+}: {
+  feature: Feature<T>;
+  shouldEvaluate: boolean;
+}): UseConditionalFeature<T> => {
+  const { user } = useContext(AuthContext);
+  const { growthbook } = useGrowthBookContext();
+  const queryKey = generateQueryKey(
+    RequestKey.Feature,
+    user,
+    feature.id,
+    growthbook,
```

Comment:
growthbook is a complex object I am not sure it would even do anything in query key since query key is stringified in react-query. Why is this needed?

Reply:
<img width="1500" alt="image" src="https://github.com/dailydotdev/apps/assets/9803078/e327c5d2-0a4c-4eb8-8c27-2757a0535e80">

this is how query key looks 😆

Reply:
Basically for [this](https://github.com/dailydotdev/apps/pull/2864#discussion_r1539387861), but not sure what alternatives we have?
There's not really a clear update/retrigger exposure I think.

Reply:
Ah, our growthbook instance never changes currently during single page visit, and we don't add new features once initiated https://github.com/dailydotdev/apps/blob/2f31655db1810b3122ca7ca20c83b0350ca4aebc/packages/shared/src/components/GrowthBookProvider.tsx#L63-L87 so it should be good to go without looking at change.

That said if it was to ever change and we change logic in `GrowthBookProvider` better way would be to just invalidate `queryClient.invalidateQueries(RequestKey.Feature)` I think.

Reply:
Good idea!

Reply:
Thats we have "you need to refresh" for experiment to become active thing on preview as well.

Reply:
Moved we'll need to check on preview as locally doesn't really work to override.